pinot-controller/src/test/java/org/apache/pinot/controller/helix/core/minion/PinotTaskManagerStatelessTest.java

```diff
@@ -203,8 +203,10 @@ public void testPinotTaskManagerSchedulerWithRestart()
         "JobGroupNames should have SegmentGenerationAndPushTask only");
     validateJob(MinionConstants.SegmentGenerationAndPushTask.TASK_TYPE, "0 */10 * ? * * *");
 
-    // Restart controller
+    // Restart controller. We need to set the port after stopping the controller because we are reusing the same config.
+    int controllerPort = _controllerPort;
     stopController();
+    _controllerPort = controllerPort;
```

Comment:
If not set like here, will the port be assigned randomly?

Reply:
No, it will through exception because it is assigned when creating the config, and reset when controller stops. With the changes in #13390 port will no longer be assigned randomly